add mention of property access on numeric literals (#37681)
mdn/content: files/en-us/web/javascript/reference/errors/identifier_after_number/index.md

const foo = life1;
```

> In JavaScript, there is a syntactic peculiarity when calling properties or methods on numbers. If you want to call a method on an integer, you cannot immediately use a dot after the number because the dot is interpreted as the start of a decimal fraction. This will result in an error. To avoid this, you need to either wrap the number in parentheses or use a double dot for a floating-point number.

Reviewer: ```suggestion
In JavaScript, there is a syntactic peculiarity when calling properties or methods on numbers. If you want to call a method on an integer, you cannot immediately use a dot after the number because the dot is interpreted as the start of a decimal fraction, causing the parser to see the method's name as an identifier immediately after a number literal. To avoid this, you need to either wrap the number in parentheses or use a double dot, where the first dot is a decimal point for the number literal and the second dot is the property accessor.
```

```js-nolint example-bad
alert(typeof 1.toString())
// SyntaxError: identifier starts immediately after numeric literal
```

Correct ways to call methods on numbers:

```js-nolint example-good
// Wrap the number in parentheses
alert(typeof (1).toString());

> // Use a floating-point number by adding an extra dot

Reviewer: ```suggestion
// Add an extra dot for the number literal
```

alert(typeof 2..toString());

// Use square brackets

> alert(typeof 3['toString']())

Reviewer: ```suggestion
alert(typeof 3["toString"]());
```
